pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +33,90 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// check if payload secret is same as SBR Secret
+func (s *secret) isSame(SBRSecret string) bool {
```

Comment:
I believe this should be a utility function that is called from the same place this is currently called, and  retrieving the secret and other related data should be the caller's perspective.

As a hard rule, always check whether the method/function name is misleading the reader, in this case it is since the name doesn't mention anything regarding the fetch.

Reply:
I have removed this function.

> retrieving the secret and other related data should be the caller's perspective

Done here ->
https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-17972c83fa5f6d6588fce52545087fd259379261db8717e2057a37c1004f7d86R303

> always check whether the method/function name is misleading the reader, in this case it is since the name doesn't mention anything regarding the fetch.

Got it.